notebooks/1.2-pyc-eda-mix.py

```diff
@@ -0,0 +1,500 @@
+from plotly.subplots import make_subplots
+import plotly.graph_objects as go
+import plotly.express as px
+import seaborn as sns
+import pandas as pd
+import numpy as np
+
+#Cdf function
+def ecdf(x):
+    x = np.sort(x)
+    def result(v):
+        return np.searchsorted(x, v, side='right') / x.size
+    return result
+
+def multiple_ecdf(column, target='SalePrice'):
+    fig = go.Figure()
+    keys = train[column].unique()
+    for key in keys:
+        bool_series = train[column]==key
+        fig.add_trace(
+            go.Scattergl(
+                x=np.unique(train[bool_series][target]), 
+                y=ecdf(train[bool_series][target])(np.unique(train[bool_series][target])), 
+                line_shape='hv',
+                name=str(key) + ', total: ' + str(bool_series.sum()))
+        )
+    return fig.show()
+
+train = pd.read_csv('../data/raw/train.csv')
+
+
+# ## Mix Columns overview
+# 
+# The columns of the training dataset that represents what we're calling mix data are represented in the strings in the variable **mix_columns**
+
+mix_columns = ['MSSubClass', 'MSZoning', 'LotFrontage', 'LotArea',
+               'Street', 'Alley', 'LotShape', 'LandContour', 'Utilities',
+               'LotConfig', 'LandSlope', 'Neighborhood', 'Condition1',
+               'Condition2', 'YearBuilt', 'YearRemodAdd', 'BldgType', 'HouseStyle',
+               'Functional', 'MiscFeature', 'MiscVal', 'MoSold', 'YrSold',
+               'SaleType', 'SaleCondition', 'SalePrice']
+
+train[mix_columns].head()
+
+
+train[mix_columns].isna().mean()
+
+
+train[mix_columns].describe()
+
+
+# # Sales variables
+# 
+# - MoSold
+# - YrSold
+# - SaleType
+# - SaleCondition
+
+# ## Sale Condition
+# 
+# 
+# Condition of sale
+# 
+#        Normal	Normal Sale
+#        Abnorml	Abnormal Sale -  trade, foreclosure, short sale
+#        AdjLand	Adjoining Land Purchase
+#        Alloca	Allocation - two linked properties with separate deeds, typically condo with a garage unit	
+#        Family	Sale between family members
+#        Partial	Home was not completed when last assessed (associated with New Homes)
+# 
+
+multiple_ecdf('SaleCondition')
+
+
+# **Insights**
+# 
+# - The partial category in this column appears to increase significantly the sale Price
+# - Maybe the other categories can be joined
+
+# ## SaleType
+# 
+# 
+# SaleType: Type of sale
+# 		
+#        WD 	Warranty Deed - Conventional
+#        CWD	Warranty Deed - Cash
+#        VWD	Warranty Deed - VA Loan
+#        New	Home just constructed and sold
+#        COD	Court Officer Deed/Estate
+#        Con	Contract 15% Down payment regular terms
+#        ConLw	Contract Low Down payment and low interest
+#        ConLI	Contract Low Interest
+#        ConLD	Contract Low Down
+#        Oth	Other
+
+multiple_ecdf('SaleType')
+
+
+# **Insights**
+# 
+# - have a lot of categories, but appears that just the category New increase in the sale
+# - Maybe we should try an unsupervised model to join these categories
```

Comment:
What did you mean with unsupervised?

Reply:
The machine learning is splited into supervised and unsupervised methods. See this [link](https://www.guru99.com/unsupervised-machine-learning.html) to understand.